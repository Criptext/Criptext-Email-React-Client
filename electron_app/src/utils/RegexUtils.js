/* eslint no-useless-escape: 0 */
const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
const HTMLTagsRegex = /<[^>]*>?/g;
const mailtoProtocolRegex = /^mailto:/;
const mailformedEventRegex = /(\[|,)null(\]|,)/g;
const percentRegex = /\d+(%)/;
const backupFilenameRegex = /Backup-([1-9]\d{3}-(0[1-9]|1[0-2])-([1-9]|[12]\d|3[01]))_(1[0-9]|2[0-3])-([0-9]\d{1}).db/i;

module.exports = {
  emailRegex,
  percentRegex,
  HTMLTagsRegex,
  mailtoProtocolRegex,
  mailformedEventRegex,
  backupFilenameRegex
};
