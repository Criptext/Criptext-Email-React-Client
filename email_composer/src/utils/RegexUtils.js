/* eslint no-useless-escape: 0 */
export const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
export const HTMLTagsRegex = /<[^>]*>?/g;
export const contactsRegex = string => {
  const escaped = escapeRegEx(string);
  return new RegExp(`(^${escaped}| ${escaped})`, 'i');
};
const escapeRegEx = s => {
  return s.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
};
