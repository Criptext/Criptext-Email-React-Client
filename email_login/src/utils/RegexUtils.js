export const usernameRegex = /(?=^([a-z0-9]([._-]{0,2}[a-z0-9])+)$)(?:^.{3,64}$)$/;
