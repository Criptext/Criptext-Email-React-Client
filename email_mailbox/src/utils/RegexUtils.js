/* eslint no-useless-escape: 0 */
export const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
export const HTMLTagsRegex = /<[^>]*>?/g;
export const mimeTypeImageRegex = /image\/([a-z][-+])*/;
export const usernameRegex = /(?=^([a-z0-9]([._-]{0,2}[a-z0-9])+)$)(?:^.{3,16}$)$/;
