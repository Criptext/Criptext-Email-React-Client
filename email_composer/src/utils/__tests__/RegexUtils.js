/* eslint-env node, jest */
import { contactsRegex } from '../RegexUtils';

const contacts = [
  'User Test1',
  'New Test 1',
  'user1@criptext.com',
  'New User2',
  'user 3',
  'newuser@criptext.com'
];

describe('Regex Utils - Filter Contacts :', () => {
  it('Should filter contacts with word beginning with user', () => {
    const filterParam = 'user';
    const filtered = contacts.filter(contact =>
      contactsRegex(filterParam).test(contact)
    );
    expect(filtered).toMatchSnapshot();
  });
});
