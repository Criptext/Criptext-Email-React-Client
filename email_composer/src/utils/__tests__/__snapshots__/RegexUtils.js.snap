// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regex Utils - Filter Contacts : Should filter contacts with word beginning with user 1`] = `
Array [
  "User Test1",
  "user1@criptext.com",
  "New User2",
  "user 3",
]
`;
